lib/logstash/outputs/file.rb

```diff
@@ -198,9 +207,27 @@ def close_stale_files
   end
 
   private
-  def open(path)
-    return @files[path] if @files.include?(path) and not @files[path].nil?
+  def cached?(path)
+     @files.include?(path) && !@files[path].nil?
+  end
 
+  private
+  def deleted?(path)
+    !File.exist?(path)
+  end
+
+  private
+  def open(path)
+    if !deleted?(path) && cached?(path)
```

Comment:
This introduces up to 2 File.stat calls _per event_ sent to this output where previously there were zero File.stat calls per event.

I wonder about the performance implications of this.

Maybe we should only bother checking if the file is deleted if `@create_if_deleted` is true, that way the performance penalty is isolated to those wishing to use this 'create-if-deleted' behavior?